gpflow/features.py

```diff
@@ -66,13 +66,64 @@ def Kuf(self, kern, Xnew):
         return Kzx
 
 
+class Multiscale(InducingFeature):
+    """
+    Multi-scale inducing features
+    Originally proposed in
+
+    ::
+
+      @incollection{NIPS2009_3876,
+        title = {Inter-domain Gaussian Processes for Sparse Inference using Inducing Features},
+        author = {Miguel L\'{a}zaro-Gredilla and An\'{\i}bal Figueiras-Vidal},
+        booktitle = {Advances in Neural Information Processing Systems 22},
+        year = {2009},
+      }
+
+    """
+    def __init__(self, Z, scales):
+        super().__init__()
+        self.Z = Param(Z)  # Multi-scale feature centres
+        self.scales = Param(scales, transform=transforms.positive)  # Multi-scale feature widths (std. dev. of Gaussian)
+        self._M = len(Z)
+
+    def __len__(self):
+        return self._M
+
+    def _cust_square_dist(self, A, B, sc):
+        return tf.reduce_sum(tf.square((tf.expand_dims(A, 1) - tf.expand_dims(B, 0)) / sc), 2)
+
+    def Kuf(self, kern, Xnew):
+        if isinstance(kern, kernels.RBF):
```

Comment:
Maybe we should depend on the multimethod package after all; this would be a good example - even if you write the feature as a class, you could have a `@multimethod` on the method, then other packages could extend this for other kernels...

Reply:
True, true. Is there a package included in Python that supports this? A quick search yields several 3rd-party packages.

Reply:
No, there is no "standard library" version, but I've used this one, which seems to do the job well, and has comparatively succinct and tidy code: https://pypi.python.org/pypi/multimethod/0.6

Reply:
Are you ok with deferring this to a later PR? I think GPflow is rightly a bit conservative about using other packages. Everybody agrees that inter-domain features should go in, so I suggest we do that first, and then do enhancement.

I do agree that it's the best solution, and I'll happily push for it to get in.

Reply:
Sure, no problem. The multimethod implementation is less than 100 lines of actual code; if necessary, we could also simply include that file..

Reply:
I think we can depend on the library. I just think we should split up these discussions a bit.